File: evidence/generated-systems/angular/hr-on-angular/src/app/models/JobRequisition.js

// Define collection and schema for JobRequisition
export interface JobRequisition {
    requisitionNumber:
	type : string
    title:
	type : string
    openings:
	type : number
    targetStartDate:
	type : Date
    Department:
	type : Schema.Types.ObjectId
    HiringManager:
	type : Schema.Types.ObjectId
    Recruiter:
	type : Schema.Types.ObjectId
    JobProfile:
	type : Schema.Types.ObjectId
    Candidates:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Candidate' }]
    Interviews:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Interview' }]
    Offers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Offer' }]
    Status:
 	type : String
    Priority:
 	type : String
#
    collection: 'jobRequisitions'
}

File: evidence/generated-systems/golang/hr-on-golang/hr-on-golang/internal/model/JobRequisition.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// JobRequisition Declaration
//==============================================================
type JobRequisition struct {
    gorm.Model
     RequisitionNumber                                    string
    Title                                    string
    Openings                                                            string
    TargetStartDate                                                            time.Time
    DepartmentId         *uint
    Department           *Department `gorm:"foreignKey:DepartmentId"`
    HiringManagerId         *uint
    HiringManager           *Employee `gorm:"foreignKey:HiringManagerId"`
    RecruiterId         *uint
    Recruiter           *Employee `gorm:"foreignKey:RecruiterId"`
    JobProfileId         *uint
    JobProfile           *JobProfile `gorm:"foreignKey:JobProfileId"`
     Candidates           []Candidate `gorm:"foreignKey:CandidatesFromJobRequisitionId"`
     Interviews           []Interview `gorm:"foreignKey:InterviewsFromJobRequisitionId"`
     Offers           []Offer `gorm:"foreignKey:OffersFromJobRequisitionId"`
    Status                      RequisitionStatus
    Priority                      RequisitionPriority

// parent associations as their child

}

